File: src/jvmMain/java/ess/papyrus/VarNull.java
package ess.papyrus;

import org.jetbrains.annotations.NotNull;

import java.nio.ByteBuffer;

/**
 * Variable that stores nothing.
 */
final public class VarNull extends Variable {

    public VarNull(@NotNull ByteBuffer input) {
        this.VALUE = input.getInt();
    }

    @Override
    public int calculateSize() {
        return 5;
    }

    @Override
    public void write(@NotNull ByteBuffer output) {
        this.getType().write(output);
        output.putInt(this.VALUE);
    }

    @NotNull
    @Override
    public VarType getType() {
        return VarType.NULL;
    }

    @NotNull
    @Override
    public String toValueString() {
        return "NULL";
    }

    @NotNull
    @Override
    public String toString() {
        return "NULL";
    }

    final private int VALUE;
}

File: src/jvmMain/kotlin/ess/papyrus/VarNull.kt
package ess.papyrus

import PlatformByteBuffer

/**
 * Variable that stores nothing.
 */
class VarNull(input: PlatformByteBuffer) : Variable() {
    override fun calculateSize(): Int {
        return 5
    }

    override fun write(output: PlatformByteBuffer?) {
        type.write(output)
        output?.putInt(VALUE)
    }

    override val type: VarType
        get() = VarType.NULL

    override fun toValueString(): String {
        return "NULL"
    }

    override fun toString(): String {
        return "NULL"
    }

    private val VALUE: Int = input.getInt()

}